07_errors/willshen8/puppy.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+type Puppy struct {
+	ID    uint32
+	Breed string
+	Color string
+	Value string
```

Comment:
Not sure if value should be numeric or a string. Most days I would argue numeric, but there are non-trivial challenges storing monetary data, and for the json api you will need to do the integer conversion anyway so I will let this slide.

On a side note, A couple of people at anz have been working on an IEEE decimal implementation in go (which solves the problems with storing dollar values). Once that is in, there would be no reason to keep using strings for dollar values.

Reply:
I think the original intention was that we could have currency signs like $.